src/components/DayPickerRangeController.jsx

```diff
@@ -163,7 +163,7 @@ export default class DayPickerRangeController extends React.Component {
     if (focusedInput !== END_DATE) return false;
 
     if (startDate) {
-      const dayDiff = day.diff(startDate, 'days');
+      const dayDiff = day.day() - startDate.day();
```

Comment:
Ugh, I meant `date()` which gets date of month and not `day()`. But that has the same issue with year boundaries. I then tried comparing both `.year()` and then comparing `.dayOfYear()`, but it seems like this is a lot less performant than the clone strategy, which surprised me. So looks like yes, that is the way to go. 

For what it's worth, I also tested skipping setting `day`, and it's quite a bit better. What do you think?

I can go ahead with whichever strategy is the most ideal.

Thanks!

<img width="977" alt="screen shot 2017-02-10 at 5 11 41 pm" src="https://cloud.githubusercontent.com/assets/5831434/22846147/1acb630c-efb4-11e6-8f38-b7f8b34cbf0c.png">